docs/developer-docs/setup/hello-world.mdx

```diff
@@ -0,0 +1,254 @@
+
+For information about project structure and default configuration, [learn more about deploying your first canister](first-canister.mdx).
+ 
+## Smart contract code
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+// src/hello_backend/main.mo
+actor {
+public query func greet(name : Text) : async Text {
+    return "Hello, " # name # "!";
+};
+};
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```rust
+// src/hello_world_backend/src/lib.rs
+#[ic_cdk::query]
+fn greet(name: String) -> String {
+    format!("Hello, {}!", name)
+}
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+```typescript
+// azle_hello/src/index.ts
+import { Canister, query, text, update, Void } from 'azle';
+
+// This is a global variable that is stored on the heap
+let message = '';
+
+export default Canister({
+    // Query calls complete quickly because they do not go through consensus
+    getMessage: query([], text, () => {
+        return message;
+    }),
+    // Update calls take a few seconds to complete
+    // This is because they persist state changes and go through consensus
+    setMessage: update([text], Void, (newMessage) => {
+        message = newMessage; // This change will be persisted
+    })
+});
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+Create the project directory and file structure:
+
+```python
+# kybra_hello/src/main.py
+
+from kybra import query, update, void
+
+# This is a global variable that is stored on the heap
+message: str = ''
+
+# Query calls complete quickly because they do not go through consensus
+@query
+def get_message() -> str:
+    return message
+
+# Update calls take a few seconds to complete
+# This is because they persist state changes and go through consensus
+@update
+def set_message(new_message: str) -> void:
+    global message
+    message = new_message # This change will be persisted
+```
+
+</TabItem>
+</Tabs>
+
+
+## Deploy locally 
+
+Deploy the project's canisters locally:
+
+```
+dfx deploy
+```
+
+## Call the canister
+
+<Tabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```bash
+dfx canister call hello_backend greet world
+```
+
+Output:
+
+```
+("Hello, world!")
+```
+
+</TabItem>
+<TabItem value="rust" label="Rust">
+
+```bash
+dfx canister call hello_backend greet world
+```
+
+Output:
+
+```
+("Hello, world!")
+```
+
+</TabItem>
+
+<TabItem value="typescript" label="TypeScript">
+
+```bash
+dfx canister call azle_hello setMessage '("Hello, world!")'
+```
+
+Then, call the `getMessage` function:
+
+```
+dfx canister call azle_hello getMessage
+```
+
+Output:
+
+```
+("Hello, world!")
+```
+
+</TabItem>
+
+<TabItem value="python" label="Python">
+
+```bash
+dfx canister call kybra_hello set_message '("Hello, world!")'
+```
+
+Then, call the `getMessage` function:
+
+```
+dfx canister call kybra_hello getMessage
+```
+
+Output:
+
+```
+("Hello, world!")
+```
+
+</TabItem>
+</Tabs>
+
+
```

Comment:
Let's add a section to show the cool feature of ICP:

## Call the smart contract from browser

It is possible to access the frontend smart contract directly from the browser. For that you need to navigate to the URL that was printed in the output of the `dfx deploy` command (your URL might differ):
```
Frontend canister via browser
    hello_frontend: http://127.0.0.1:4943/?canisterId=bd3sg-teaaa-aaaaa-qaaba-cai
```

This will render a webpage with a form that accepts a string and sends it to the backend smart contract.

The source code of the webpage is defined in `src/hello_frontend/src/index.{html,js}` files.
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <title>hello</title>
    <base href="/" />
    <link rel="icon" href="favicon.ico" />
    <link type="text/css" rel="stylesheet" href="main.css" />
  </head>
  <body>
    <main>
      <img src="logo2.svg" alt="DFINITY logo" />
      <br />
      <br />
      <form action="#">
        <label for="name">Enter your name: &nbsp;</label>
        <input id="name" alt="Name" type="text" />
        <button type="submit">Click Me!</button>
      </form>
      <section id="greeting"></section>
    </main>
  </body>
</html>
```

```js
import { hello_backend } from "../../declarations/hello_backend";

document.querySelector("form").addEventListener("submit", async (e) => {
  e.preventDefault();
  const button = e.target.querySelector("button");

  const name = document.getElementById("name").value.toString();

  button.setAttribute("disabled", true);

  // Interact with foo actor, calling the greet method
  const greeting = await hello_backend.greet(name);

  button.removeAttribute("disabled");

  document.getElementById("greeting").innerText = greeting;

  return false;
});
```

The JavaScript code running in browser uses the [agent-js](https://github.com/dfinity/agent-js/tree/main) library to interact with the backend smart contract. The library creates a wrapper JavaScript object based on the interface of the smart contract. That wrapper object provides convenience methods for calling the smart contract functions as if they were JavaScript functions:
```js
const greeting = await hello_backend.greet(name);
```